### PR TITLE
removed @chakra-ui dependencies

### PR DESCRIPTION
**Pull request by ehconitin**
Issue #6976 
@FelixMalfait 

I could not do
```
import { Banner } from 'twenty-ui';

const StyledBanner = styled(Banner)
  display: flex;
  align-items: center;
  padding: ${({ theme }) => theme.spacing(8)};
  position: absolute;
  border-radius: 8px;
  &:hover {
    background-color: ${({ theme }) => theme.accent.primary};
  }
;
```
The styles wont get overridden for Banner, so for now I styled a new banner in `UnmatchColumnBanner` which is inconsistent.
I couldnt figure out why css properties are not being overridden, need help!

@Bonapara 
Question - 
Should the click work on entire banner or just cheveron? For now it just on cheveron click.

https://github.com/user-attachments/assets/0f409e78-a341-4f26-af74-117e4b2775a9

### CODE REVIEW

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx [hunk 1]
```diff
@@ -70,42 +27,70 @@ type UnmatchColumnProps<T extends string> = {
...
-        </Accordion>
-      </StyledAccordionContainer>
-    )
+    <StyledContainer>
+      <UnmatchColumnBanner
+        message={getAccordionTitle(fields, column)}
+        buttonOnClick={() => setIsExpanded(!isExpanded)}
+        isExpanded={isExpanded}
+      />
+      <StyledTransitionContainer isExpanded={isExpanded} height={contentHeight}>
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx:81]: 👍🏻

File: package.json [hunk 2]
```diff
@@ -8,8 +8,6 @@
     "@aws-sdk/credential-providers": "^3.363.0",
     "@blocknote/mantine": "^0.15.3",
     "@blocknote/react": "^0.15.3",
-    "@chakra-ui/accordion": "^2.3.0",
```
Comment (charlesBochet) [inline package.json:11]: love it!

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx [hunk 3]
```diff
@@ -70,42 +27,70 @@ type UnmatchColumnProps<T extends string> = {
...
+
+const StyledTransitionContainer = styled.div<{
+  isExpanded: boolean;
+  height: number;
+}>`
+  max-height: ${({ isExpanded, height, theme }) =>
+    isExpanded ? `calc(${height}px + ${theme.spacing(4)})` : '0'};
+  overflow: hidden;
+  position: relative;
+  transition: max-height 0.4s ease;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx:43]: we have transition variable in theme

Comment (charlesBochet) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx:43]: It's actually under animation.ts, please use 'normal' value

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx [hunk 4]
```diff
@@ -70,42 +27,70 @@ type UnmatchColumnProps<T extends string> = {
...
 }: UnmatchColumnProps<T>) => {
   const { fields } = useSpreadsheetImportInternal<T>();
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [contentHeight, setContentHeight] = useState(0);
+  const contentRef = useRef<HTMLDivElement>(null);
 
   const column = columns[columnIndex];
   const isSelect = 'matchedOptions' in column;
 
+  useLayoutEffect(() => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx:66]: what is the purpose of this layoutEffect exactly?

Looking at the code I understand it is only to set the TransitionContainer height. If yes, I think we should extract this to a re-usable component like ExpandableContainer and hide this logic in the component

Comment (charlesBochet) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx:66]: We could add this component to twenty-ui so it's part of the toolbox

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumnBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import { IconChevronDown, IconInfoCircle } from 'twenty-ui';
+const StyledBanner = styled.div`
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumnBanner.tsx:4]: Okay, so you don't need to duplicate the Banner code here :)

To answer your question, you need to add "className" as a prop of the Banner to be able to "Styled" it through styled-component. Just adding this props will allow you to use `styled(Banner)` syntax

**Review by charlesBochet — COMMENTED** [review]
Love it @ehconitin! I have added comments and answered your question :)

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx
Comment (ehconitin) [inline packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx:1]: What would be the best place to keep this component in?
Also, if we’re putting it in twenty-ui, should I add a button in the Storybook story to toggle the expand/collapse state? I’d need to use a state variable for that. Is that okay?

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx (hunk 4 above)
Comment (ehconitin) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx]: Yes. The useLayoutEffect is just there to calculate the container’s height after the DOM updates so the TransitionContainer can adjust its height correctly.

I’ve extracted this logic into a new component, ExpandableContainer.  Adding it to twenty-ui sounds like a great idea.

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumnBanner.tsx (hunk 5 above)
Comment (ehconitin) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumnBanner.tsx]: Ah, I am so dumb! haha 
Thanks :)

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx
Comment (charlesBochet) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx:1]: yes good idea!
You can wrap your ExpandableContainer in a component in the story, we have many examples of that in the codebase

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { ANIMATION, isDefined } from 'twenty-ui';
+
+const StyledTransitionContainer = styled.div<{
+  isExpanded: boolean;
+  height: number;
+}>`
+  max-height: ${({ isExpanded, height }) => (isExpanded ? `${height}px` : '0')};
+  overflow: hidden;
+  position: relative;
+  transition: max-height ${ANIMATION.duration.normal}s
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx:12]: use { theme }

File: packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx
Comment (charlesBochet) [reply packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/ExpandableContainer.tsx:1]: in twenty-ui/src/layout (new folder) I would say

**Review by charlesBochet — APPROVED** [review]
Bravo!

**Pull request merged**